server/chat-plugins/customavatars.js

```diff
@@ -0,0 +1,106 @@
+//Original plugin from https://github.com/CreaturePhil/Showdown-Boilerplate/blob/master/chat-plugins/customavatar.js.
+//Credits to CreaturePhil and the other listed contributors.
+//updated for the main server by Maxalexanderpi, with help from Hoeenhero.
+'use strict';
+
+/** @type {typeof import('../../lib/fs').FS} */
+const FS = require(/** @type {any} */('../../.lib-dist/fs')).FS;
+
+const path = require('path');
+const fs = require('fs');
+
+/* eslint no-restricted-modules: [0] */
+const AVATAR_PATH = 'config/avatars/';
+
+const VALID_EXTENSIONS = ['.png'];
+
+const https = require('https');
+
+function downloadImage(image_url, name, extension) {
+	let req = https.get(image_url, res => {
+		 res.setEncoding('binary');
+		 res.on('response', response => {
+			 if (response.statusCode !== 200) return;
+			 let type = response.headers['content-type'].split('/');
+			 if (type[0] !== 'image') return;
+
+			response.pipe(FS(AVATAR_PATH + name + extension).createWriteStream());
+		 });
+	});
+	req.on('error', e => {
+		console.error(e);
+	});
+	req.end();
+}
+function loadCustomAvatars() {
+	fs.readdir(AVATAR_PATH, (err, files) => {
+		if (err) console.log("Error loading custom avatars: " + err);
+		if (!files) files = [];
+		files
+			.filter(file => VALID_EXTENSIONS.includes(path.extname(file)))
+			.forEach(file => {
+				let name = path.basename(file, path.extname(file));
+				Config.customavatars[name] = file;
+			});
+	});
+}
+
+loadCustomAvatars();
+
+exports.commands = {
+	customavatar: {
+		add(target, room, user) {
+			if (!this.can('avatar')) return false;
```

Comment:
I don't think `avatar` is a permission in `config/config.js`?
```suggestion
			if (!this.can('declare')) return;
```